dash-pipeline/bmv2/dash_outbound.p4

```diff
@@ -5,212 +5,197 @@
+        meta.tunnel_0.tunnel_dip = tunnel_dip != 0 ? tunnel_dip : meta.tunnel_0.tunnel_dip;
+        meta.tunnel_0.tunnel_smac = tunnel_smac != 0 ? tunnel_smac : meta.tunnel_0.tunnel_smac;
+        meta.tunnel_0.tunnel_dmac = tunnel_dmac != 0 ? tunnel_dmac : meta.tunnel_0.tunnel_dmac;
+
+        meta.tunnel_source = tunnel_source != 0 ? tunnel_source : meta.tunnel_source;
+        meta.tunnel_target = tunnel_target != 0 ? tunnel_target : meta.tunnel_target;
+        meta.tunnel_underlay0_id = tunnel_underlay0_id != 0 ? tunnel_underlay0_id : meta.tunnel_underlay0_id;
+        meta.tunnel_underlay1_id = tunnel_underlay1_id != 0 ? tunnel_underlay1_id : meta.tunnel_underlay1_id;
     }
 
-#ifdef TARGET_BMV2_V1MODEL
-
-    direct_counter(CounterType.packets_and_bytes) routing_counter;
-#endif // TARGET_BMV2_V1MODEL
-#ifdef TARGET_DPDK_PNA
-#ifdef DPDK_SUPPORTS_DIRECT_COUNTER_ON_WILDCARD_KEY_TABLE
-    // See the #ifdef with same preprocessor symbol in dash_pipeline.p4
-    DirectCounter<bit<64>>(PNA_CounterType_t.PACKETS_AND_BYTES) routing_counter;
-#endif  // DPDK_SUPPORTS_DIRECT_COUNTER_ON_WILDCARD_KEY_TABLE
-#endif  // TARGET_DPDK_PNA
-
-    @name("outbound_routing|dash_outbound_routing")
-    table routing {
+    @SaiTable[name = "outbound_routing", stage = "routing0", api = "dash_outbound_routing"]
+    table routing0 {
         key = {
-            meta.eni_id : exact @name("meta.eni_id:eni_id");
-            meta.is_overlay_ip_v6 : exact @name("meta.is_overlay_ip_v6:is_destination_v4_or_v6");
-            meta.dst_ip_addr : lpm @name("meta.dst_ip_addr:destination");
+            meta.pipeline_oid : exact @SaiVal[type = "sai_object_id_t"];
+            meta.pkt_meta.lookup_addr_is_v6 : exact;
+            meta.pkt_meta.lookup_addr : lpm;
         }
 
         actions = {
-            route_vnet; /* for expressroute - ecmp of overlay */
-            route_vnet_direct;
-            route_direct;
-            route_service_tunnel;
+            outbound_metadata_publish;
             drop;
         }
         const default_action = drop;
-
-#ifdef TARGET_BMV2_V1MODEL
-        counters = routing_counter;
-#endif // TARGET_BMV2_V1MODEL
-#ifdef TARGET_DPDK_PNA
-#ifdef DPDK_SUPPORTS_DIRECT_COUNTER_ON_WILDCARD_KEY_TABLE
-        pna_direct_counter = routing_counter;
-#endif // DPDK_SUPPORTS_DIRECT_COUNTER_ON_WILDCARD_KEY_TABLE
-#endif // TARGET_DPDK_PNA
     }
 
-    action set_tunnel(IPv4Address underlay_dip,
-                      bit<16> meter_class,
-                      bit<1> meter_class_override) {
-        meta.encap_data.underlay_dip = underlay_dip;
-        meta.mapping_meter_class = meter_class;
-        meta.mapping_meter_class_override = meter_class_override;
-        meta.encap_data.dash_encapsulation = dash_encapsulation_t.VXLAN;
-    }
+    @SaiTable[name = "outbound_routing", stage = "routing1", api = "dash_outbound_routing"]
+    table routing1 {
```

Comment:
let's remove the second stage for routing and mapping, so we can focus and get the minimum case working for sai header generation.